### ISSUE TITLE
Performance problems, Memory Leaks

### ISSUE THREAD

**Issue by marlon360** [marlon360/rki-covid-api#305]
![Bildschirmfoto 2021-11-01 um 11 59 34](https://user-images.githubusercontent.com/11810704/139661845-6e468053-3f43-48a8-b251-a184a418a3eb.png)

This chart shows the CPU usage and the memory consumption on the server.
The dotted line marks a server restart.

The memory consumption increases over time, which indicated a memory leak and missing garbage collection.
Furthermore the CPU spikes to 100% from time to time.

Help is really appreciated to eliminate these problems.

**Comment by Rubber1Duck** [comment]
Hi Marlon, maybe thats the rediscache.When the rediscache is starting up, there is a Report that a configuration is missing. I dont know the exact report, and i m on vaccation in the moment.....Maybe you should check this.....Am 01.11.2021 12:05 schrieb Marlon Lückert ***@***.***>:

This chart shows the CPU usage and the memory consumption on the server.
The dotted line marks a server restart.
The memory consumption increases over time, which indicated a memory leak and missing garbage collection.
Furthermore the CPU spikes to 100% from time to time.
Help is really appreciated to eliminate these problems.

**Comment by Rubber1Duck** [comment]
The Message what i meen is this

WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.

**Comment by marlon360** [comment]
The memory usage of Redis is fine:
![Bildschirmfoto 2021-11-12 um 11 00 54](https://user-images.githubusercontent.com/11810704/141448621-25d6b9f9-777a-41cd-b1a5-518611cf5646.png)

There are some large objects that are held in memory. But it is really hard to debug that problem.

**Comment by Rubber1Duck** [comment]
Did you have this Issue as bevor you switched to railway, too?

**Comment by marlon360** [comment]
Yes I think so. But the old server was a fixed price with 32GB RAM included and with Railway it is pay-per-usage, which means I am paying for every single MB of RAM that is currently used. So we can reduce costs with every MB we save.

In front of the railway server is a cloudflare cdn, where I cache the requests. So most of the traffic is actually not hitting the server:
![1F5B7EBE-0A3E-40BE-A3EA-E1519EC50A4D](https://user-images.githubusercontent.com/11810704/141478596-8ef13c68-b1df-4c8f-b1d2-529a435d8f46.jpeg)
